bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/graphics/GC.java

```diff
@@ -466,7 +466,7 @@ void copyAreaInPixels(Image image, int x, int y) {
 	if (image == null) SWT.error(SWT.ERROR_NULL_ARGUMENT);
 	if (image.type != SWT.BITMAP || image.isDisposed()) SWT.error(SWT.ERROR_INVALID_ARGUMENT);
 	/* Copy the bitmap area */
-	Rectangle rect = image.getBoundsInPixels();
+	Rectangle rect = image.getBounds(data.nativeZoom);
```

Comment:
This modifies the behavior, doesn't it?
Let's assume `getBoundsInPixels()` returned `(100, 100, 100, 100)` and let's assume that `data.nativeZoom=125`. Then `getBounds(data.nativeZoom)` will scale up from 100 to 125, i.e., it will return `(125, 125, 125, 125)`. Do I miss something?

Reply:
Depends what the native zoom of the image is. If the native zoom of the image is 125, then it will return (100, 100, 100, 100). If the native zoom of the image is 100, then yes. Probably would be could to look out for a snippet, where this is used and play around with that.

Reply:
I do not understand that. Why should it return (100, 100, 100, 100) when the native zoom of the image is 125? Within `getBounds(int)`, the "corrected" zoom is used according to `DPIUtil.getZoomForAutoscaleProperty(initialNativeZoom)`, so even if the native zoom of the image is 125, a value of 100 is used for scaling, isn't it?

Reply:
So you mean the call should be getBounds(DPI.Util.getZoomForAutoscaleProperty(data.nativeZoom)) to be in the same zoom restrictions as Image uses internally?

Reply:
What I meant for example is the scenario was with mode exact or quarter, where the Image was e.g. created on the primary monitor with 125%. Then getBounds(125) would be equal to getZoom() from the image, so it would return  (100, 100, 100, 100)

Reply:
That's true. I had the "default" case in mind (i.e., scaling mode `integer200`) in which I still think that the behavior might change.

Reply:
> That's true. I had the "default" case in mind (i.e., scaling mode `integer200`) in which I still think that the behavior might change.

You are right, the behavior changes. I tested it with the snippet 215, where image.getBoundsInPixels() returns Rectangle {0, 0, 5120, 1440} but later scaled due to native zoom (which is 200) to final value  image.getBounds(data.nativeZoom) : Rectangle {0, 0, 10240, 2880}